criu/mem.c

```diff
@@ -402,7 +408,14 @@ static int generate_vma_iovs(struct pstree_item *item, struct vma_area *vma, str
 			has_parent = false;
 	}
 
-	if (vma_entry_is(vma->e, VMA_AREA_AIORING)) {
+	/*
+	 * We want to completely ignore these VMA types on the pre-dump:
+	 * 1. VMA_AREA_AIORING because it non soft-dirty trackable (kernel writes)
+	 * 2. MAP_HUGETLB mappings because they are not premapped, so we restore it
+	 * content from the parasite context and can use only full memory image
```

Comment:
*mappings because they are not pre-mapped and we can't use parent images from pre-dump stages. Instead, the content is restored from the parasite context using full memory image.